src/api/mod.rs

```diff
@@ -320,6 +305,52 @@ impl SwClient {
         };
         result
     }
+
+    fn handle_authenticated_request(
+        &self,
+        request_builder: RequestBuilder,
+        measure_time: bool,
+    ) -> Result<(Response, Option<Duration>), SwApiError> {
```

Comment:
If you really want to keep the logging up to the caller of this function, I would still suggest removing the `measure_time` argument and `Option` around the returned duration. Just measure the duration on every call and the caller can throw that away if he doesn't need it.

```suggestion
    fn handle_authenticated_request(
        &self,
        request_builder: RequestBuilder,
    ) -> Result<(Response,Duration), SwApiError> {
```

The compiler might even be smart enough to not do it if the result isn't used anyways, and even if not, that measuring code shouldn't cause performance issues🤷

Reply:
no, I wasn't finished with my changes yet. To be honest, I don't really have an opinion on the logging.... 🙈 If its "simpler" this way, we do it this way.